test/test_runner.go

```diff
@@ -579,33 +645,33 @@ func (r *TestRunner) interpreterContractValueHandler(
 			return contract
 
 		default:
-			if _, ok := compositeType.Location.(common.AddressLocation); ok {
-				invocation, found := contractInvocations[compositeType.Identifier]
-				if !found {
-					panic(fmt.Errorf("contract invocation not found"))
-				}
-				parameterTypes := make([]sema.Type, len(compositeType.ConstructorParameters))
-				for i, constructorParameter := range compositeType.ConstructorParameters {
-					parameterTypes[i] = constructorParameter.TypeAnnotation.Type
-				}
-
-				value, err := inter.InvokeFunctionValue(
-					constructorGenerator(common.Address{}),
-					invocation.ConstructorArguments,
-					invocation.ArgumentTypes,
-					parameterTypes,
-					invocationRange,
+			var storedValue interpreter.Value
+
+			switch location := compositeType.Location.(type) {
+			case common.AddressLocation:
+				storageMap := r.storage.GetStorageMap(
+					location.Address,
+					runtime.StorageDomainContract,
```

Comment:
I think that we cannot cache the `env` variable, because the `*runtime.interpreterEnvironment` type is not exported, so I cannot use this type, even though it is the return type of `runtime.NewBaseInterpreterEnvironment(config)`.
So it does not suffice to export the `loadContract()` method. Exporting this type seems like a big change. What do you suggest @SupunS ?

Reply:
We could expose the method via an interface, and use the interface type here. For e.g: the `interpreterEnvironment ` already implements several other interface like `Environment`, `stdlib.CurrentBlockProvider`, etc. so that `interpreterEnvironment` be used as any of those.

The reason for the suggestion was to avoid having to duplicate this logic again here.

But yeah, it would need some refactoring, and I agree, might be good to do that separately. And shouldn't block this PR. 👍 
I'll also have a look sometime later.

Reply:
Makes total sense to re-use as much as possible from the already present implementations on Cadence side. Didn't think about adding a lean interface simply for the `loadContract` method, that's a good suggestion :ok_hand:

Reply:
Can you maybe create a follow-up issue to refactor?

Reply:
https://github.com/onflow/cadence/issues/2824 I have opened this issue here, on the Cadence side.